dspace-api/src/main/java/org/dspace/content/Entity.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content;
+
+import java.util.List;
+
+/**
+ * This class represents an Entity object. An Entity object has an Item that it describes with a list
+ * of relationships that it includes as well.
+ */
+public class Entity {
```

Comment:
This is currently a utility class to easily work with the functions specific to the relations.
Would you recommend to:
* change this in the docs
* Make the Entity a subclass of Item
* Move the methods to the Item
I think this is all possible, and the impact is primarily on making it easy to understand the concept

Reply:
Maybe just rename it `EntityRelationships` or the like.  That, and a prominent notice somewhere that in the current implementation "entities" are inferred, not persisted, might be enough to address confusion such as mine.

I had a very different idea of how the implementation would be organized, and it's just taking some effort to wrap my mind around the reality.  I thought perhaps I would not be alone in this.

Reply:
I think @mwoodiupui 's suggestion to rename it would also be a very good solution. This is primarily a utility class which keeps track of all the relations of a single item. The suggestion `EntityRelationships` or perhaps `EntityRelationshipsUtil` would sound good to me.

I did also verify in more detail whether we rather want to move this to the `Item` class, but I wouldn't know the best way to populate the relationships when creating an Item. In this case I would prefer to populate the relationships similar to the metadatavalues or resourcePolicies but this may be a bit more complicated.

Reply:
@benbosman : I like the suggestion to rename it. However, as noted in the [last Entities WG meeting (Nov 6)](https://wiki.duraspace.org/display/DSPACE/2018-11-06+DSpace+7+Entities+WG+Meeting), we should consider dropping the term "Entity" altogether.  This could be called `ItemRelationships` or `ItemRelationshipsUtil` , as it clarifies that these relationships are applied between `Item` objects (as we have no actual object named `Entity`).  Similarly, we should consider renaming the `entity_type` table (and all associated Java objects) to be `item_type`.  If we do not have an object named `Entity`, then we should drop all usages of that word from our codebase in order to avoid confusion.

Reply:
Thanks all for your input, we'll adjust this in a new PR to ensure this PR can be merged already